src/scenes/Engagement/InternshipEngagement/Create/CreateInternshipEngagement.tsx

```diff
@@ -0,0 +1,81 @@
+import { useSnackbar } from 'notistack';
+import React from 'react';
+import { Except } from 'type-fest';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../../components/Dialog/DialogForm';
+import { SubmitError } from '../../../../components/form';
+import { UserField, UserLookupItem } from '../../../../components/form/Lookup';
+import { ButtonLink } from '../../../../components/Routing';
+import { ProjectOverviewDocument as ProjectOverview } from '../../../Projects/Overview/ProjectOverview.generated';
+import { useCreateInternshipEngagementMutation } from './CreateInternshipEngagement.generated';
+
+interface CreateInternshipEngagementFormValues {
+  intern: UserLookupItem;
+}
+
+type CreateInternshipEngagementProps = Except<
+  DialogFormProps<CreateInternshipEngagementFormValues>,
+  'onSubmit'
+> & {
+  projectId: string;
+};
+
+export const CreateInternshipEngagement = ({
+  projectId,
+  ...props
+}: CreateInternshipEngagementProps) => {
+  const [createEngagement] = useCreateInternshipEngagementMutation();
+  const { enqueueSnackbar } = useSnackbar();
+  const submit = async (input: CreateInternshipEngagementFormValues) => {
+    const createInternshipEngagementInput = {
+      engagement: { projectId, internId: input.intern.id },
+    };
+    const res = await createEngagement({
+      variables: { input: createInternshipEngagementInput },
+      refetchQueries: [
+        {
+          query: ProjectOverview,
+          variables: { input: projectId },
+        },
+      ],
+    });
+
+    const engagement = res.data!.createInternshipEngagement.engagement;
+
+    enqueueSnackbar(
+      `Created engagement for: ${engagement.intern.value?.fullName}`,
+      {
+        variant: 'success',
+        action: () => (
+          <ButtonLink
+            color="inherit"
+            to={`projects/${projectId}/engagements/${engagement.id}`}
```

Comment:
For some reason, relative linking isn't working. /projects, /languages, /organizations all work, but if it's compounded, it doesn't.

Reply:
This is how it's done in the list item card.

https://github.com/SeedCompany/cord-field/blob/7b36305555cc9d06dfd2b50c1a59d7d290777f4f/src/components/InternshipEngagementListItemCard/InternshipEngagementListItemCard.tsx#L83-L86

Reply:
The list item card is in components/ and can be used anywhere so it's important we link to the right spot from an unknown current location. Since this is a scene it's ok to assume the current location.
I'm not sure why the relative path isn't working.
If you wanted to just change it to be absolute I would be fine with that.

Reply:
I did see this issue. 

https://github.com/ReactTraining/react-router/issues/7128

Reply:
Yeah that only affects the like "starting path" though. Relative paths still work just without a global "prefix", if you will.

Reply:
Ok. It seems like the router isn't picking up the current pathname for some reason. If it's at the root route `/`, that works fine though.

Reply:
Yes that's what I mean by absolute path 😄

Reply:
Right right, but I wanted to figure out what the problem was. Maybe not worth it

Reply:
Oh it's because it's in a snackbar which is mounted outside of this matched route.

Also, do we even need this snackbar message? the new item is displayed right below the create button.

Reply:
ah. Well. It does take a few seconds for it to show up and the other snackbars set a precedent for creates.

Reply:
The global creates are that way because the result isn't directly shown on screen. If the result is shown directly on screen, then snackbars don't add value.

If you set `awaitRefetchQueries: true` then it will show the creating spinner until the the project overview page refetches.

Reply:
Yeah. That's clean. Very slow atm though.

Reply:
Yeah I know 😕  That's not our problem though 😉